src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -222,7 +223,9 @@ def __call__(self, poller):
                 correlation_message = 'Correlation ID: {}'.format(correlation_id)
             except:  # pylint: disable=bare-except
                 pass
-            if is_verbose:
+
+            if is_verbose and datetime.datetime.now() - self.last_progress_report >= datetime.timedelta(seconds=10):
+                self.last_progress_report = datetime.datetime.now()
```

Comment:
Capture the now once instead of fetching it twice, otherwise, there is a gap between them.
```
now = datetime.datetime.now()
if is_verbose and now - self.last_progress_report >= datetime.timedelta(seconds=10):
    self.last_progress_report = now
```